runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -245,20 +246,35 @@ ILogData read(long address) {
             //    by thread A)
             ILogData data = readCache.getIfPresent(address);
             if (data == null) {
-                // Loading a value without the cache loader can result in
-                // redundant loading calls (i.e. multiple threads try to
-                // load the same value), but currently a redundant RPC
-                // is much cheaper than the cost of a NoRollBackException, therefore
-                // this trade-off is reasonable
                 final ILogData loadedVal = fetch(address);
                 data = readCache.asMap().computeIfAbsent(address, (k) -> loadedVal);
-                return data;
-            } else {
-                return data;
             }
-        } else {
-            return fetch(address);
+
+            return data;
         }
+
+        return fetch(address);
+    }
+
+    /**
+     * Read a range of addresses in a predictive way.
+     *
+     * If there is a cache miss, read a range of addresses, otherwise
+     * directly return the requested entry and read is defer.
+     */
+    public @Nonnull ILogData predictiveReadRange(Long address, List<Long> addresses) {
```

Comment:
I named it predictive because it is bringing data in advance that you know you'll need later.. It takes advantage of a cache miss and reads a batch if not present. I've added some comments, let me know if that solves the confusion or we can rename it, any suggestion?